### ISSUE TITLE
Question: Is ProteinBertModel a Devlin 2018 BERT trained on Pfam?

### ISSUE THREAD

**Issue by mRcSchwering** [songlab-cal/tape#39]
Hi, I got a little bit confused. Maybe you already mentioned it and I didn't find it.

As far as I understand the pre-trained `ProteinBertModel` (_Transformer_) is from [huggingface](https://huggingface.co/), so it is the model described by [Devlin 2018](https://arxiv.org/abs/1810.04805).

In your paper [Evaluating Protein Transfer Learning with TAPE](https://papers.nips.cc/paper/9163-evaluating-protein-transfer-learning-with-tape.pdf) however the _Transformer_ is a model as described by [Vaswani 2017](https://arxiv.org/abs/1706.03762).

So, you trained the [Devlin 2018](https://arxiv.org/abs/1810.04805)-Transformer again on Pfam as you did before for the [Vaswani 2017](https://arxiv.org/abs/1706.03762)-Transformer?

Best
Marc

**Comment by nickbhat** [comment]
We trained a Transformer Encoder using masked language modeling. The architecture is of the encoder (as well as the decoder, which we do not use) is from Vaswani et al, and the idea to pair that architecture with masked language modeling is from Devlin et al. 

We found it more clear to just cite the architecture rather than BERT for a few reasons. Two major ones 
1. There is more to BERT than just the Transformer encoder + masked language modeling. For example, they introduce an "is next task" (using cls and sep tokens), which we did not end up reporting in the TAPE paper. 
2. Our choice of hyperparameters used in the paper differs from that of BERT-base, which would've made calling our model "BERT" confusing.

But just to be clear, the Transformer architecture is the same in both papers up to hyperparameter choices (and perhaps some details of positional encoding or other minutiae). 

Does this answer your question? Let us know if you have further questions.

**Comment by datavisyn-ro-b** [comment]
Hi, thanks for the quick answer. Yes, that's what I wanted to know.

I was thinking that - intuitively - training with the Cloze task would make the transformer take the whole sequence into account. If it would have been trained like in Vaswani et al I would consider presenting both directions of an amino acid sequence (C->N and N->C).

**Comment by mRcSchwering** [comment]
Hi, thanks for the quick answer. Yes, that's what I wanted to know.

I was thinking that - intuitively - training with the Cloze task would make the transformer take the whole sequence into account. If it would have been trained like in Vaswani et al I would consider presenting both directions of an amino acid sequence (C->N and N->C).
